NitroxClient/Communication/Packets/Processors/PDAScanFinishedProcessor.cs

```diff
@@ -14,6 +14,10 @@ public override void Process(PDAScanFinished packet)
         {
             StoryManager.ScanCompleted(packet.Id, packet.Destroy);
         }
+        if (packet.WasAlreadyResearched)
```

Comment:
This is kinda overloading the packet.  is it possible that we can just send an EntityDestroyedByClient instead of PDAScanFinished in this case?  Then you don't need to add logic everwhere to handle this flag.

Reply:
The (NitroxClient) ``StoryManager.ScanCompleted`` behaviour needs to be executed for entities destroyed by this way. And we don't want it to be ran each time an entity is destroyed because most of them have nothing to do with entity scanning.
So either we share this packet's utility (which could be good because the server processor has shared behaviour for both cases) or we make another packet that makes it clear that an entity was destroyed by being scanned.

Reply:
For my education...

It looks like the client scan completed removes fragments and cachedProgress.  Wouldn't fragments already be removed on previous packets before WasAlreadyResearched is applicable?  Also, does it even matter if cached progress for the id remains?  Just trying to understand the reason that we need to execute ScanCompleted logic.